src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -663,8 +663,8 @@ public function tempnam($dir, $prefix)
     /**
      * Atomically dumps content into a file.
      *
-     * @param string $filename The file to be written to
-     * @param string $content  The data to write into the file
+     * @param string                $filename The file to be written to
+     * @param string|resource|array $content  The data to write into the file
```

Comment:
see https://github.com/symfony/symfony/pull/28019#issuecomment-410990560

im not sure anymore what's right or wrong :) but AFAIK we still want / advertise with "only string" support

raises the question if this issue conceptually applies still.

Reply:
oh, 4.x is already patched: https://github.com/symfony/symfony/pull/29661

Reply:
Yup, nevertheless it still supports `string|resource`

Reply:
To clarify the issue, the state right now (before this PR):

* 3.4: docblock advertises string, accepts string|resource|array in reality
* 4.4: docblock advertises string|resource, accepts string|resource|array in reality but array triggers deprecation warning
* 5.0: docblock advertises string|resource, accepts string|resource|array in reality but array triggers deprecation warning
* 5.1: docblock advertises string|resource, accepts string|resource|array in reality but array triggers deprecation warning

After this PR is merged the world should look like this:

* 3.4: docblock advertises string|resource|array, accepts string|resource|array in reality
* 4.4: docblock advertises string|resource, accepts string|resource|array in reality but array triggers deprecation warning
* 5.0: docblock advertises string|resource, accepts string|resource|array in reality but array triggers deprecation warning
* 5.1: docblock advertises string|resource, accepts string|resource|array in reality but array triggers deprecation warning

Reply:
yes but start advertising `|resource|array` in a patch is basically a new feature

Reply:
Well, I would say that the explicit deprecation of array usage + the introduction of `string|resource` in 4.4 recognizes this feature as "it has always been there but it was not documented". Therefore this change follows that spirit.

Reply:
advertizing `array` should be removed to me

Reply:
Done